### <a name="change-in-behavior-for-taskwaitall-methods-with-time-out-arguments"></a><span data-ttu-id="6da5e-101">タイムアウト引数を持つ Task.WaitAll メソッドの動作を変更します。</span><span class="sxs-lookup"><span data-stu-id="6da5e-101">Change in behavior for Task.WaitAll methods with time-out arguments</span></span>

|   |   |
|---|---|
|<span data-ttu-id="6da5e-102">説明</span><span class="sxs-lookup"><span data-stu-id="6da5e-102">Details</span></span>|<span data-ttu-id="6da5e-103">Task.WaitAll 動作が行われたより一貫性のある .NET 4.5.In で .NET Framework 4 で、これらのメソッドの動作が矛盾していました。</span><span class="sxs-lookup"><span data-stu-id="6da5e-103">Task.WaitAll behavior was made more consistent in .NET 4.5.In the .NET Framework 4, these methods behaved inconsistently.</span></span> <span data-ttu-id="6da5e-104">タイムアウトの期限が切れたときに、メソッドを呼び出す前に完了した、またはキャンセルされたタスクが 1 つ以上ある場合、メソッドでは <xref:System.AggregateException?displayProperty=name> 例外がスローされていました。</span><span class="sxs-lookup"><span data-stu-id="6da5e-104">When the time-out expired, if one or more tasks were completed or canceled before the method call, the method threw an <xref:System.AggregateException?displayProperty=name> exception.</span></span> <span data-ttu-id="6da5e-105">タイムアウトの期限が切れると、メソッド呼び出しの前に完了またはキャンセルされたタスクはなかったが、メソッド呼び出し後に 1 つ以上のタスクがこれらの状態になると、メソッドは false を返しました。</span><span class="sxs-lookup"><span data-stu-id="6da5e-105">When the time-out expired, if no tasks were completed or canceled before the method call, but one or more tasks entered these states after the method call, the method returned false.</span></span><br/><br/><span data-ttu-id="6da5e-106">.NET Framework 4.5 でこれらのメソッド オーバー ロードを返します、タイムアウトの期限が切れたし、スローされるときに、すべてのタスクが実行中の場合は false、 <xref:System.AggregateException?displayProperty=name> (メソッドの前後であったかに関係なく、入力タスクが取り消された場合にのみ例外呼び出す) その他のタスクがまだ実行されていないとします。</span><span class="sxs-lookup"><span data-stu-id="6da5e-106">In the .NET Framework 4.5, these method overloads now return false if any tasks are still running when the time-out interval expired, and they throw an <xref:System.AggregateException?displayProperty=name> exception only if an input task was cancelled (regardless of whether it was before or after the method call) and no other tasks are still running.</span></span>|
|<span data-ttu-id="6da5e-107">提案される解決策</span><span class="sxs-lookup"><span data-stu-id="6da5e-107">Suggestion</span></span>|<span data-ttu-id="6da5e-108">場合、<xref:System.AggregateException?displayProperty=name>コード IsCanceled プロパティ経由で同じ検出を行う必要があります代わりに呼び出される WaitAll 呼び出しの前にキャンセルされたタスクを検出するための手段としてキャッチされているされました (例: です。Any(t =&gt; t.IsCanceled)) ため、待機中のすべてのタスクがタイムアウトする前に完了した場合、.NET 4.6 はその場合にスローのみです。</span><span class="sxs-lookup"><span data-stu-id="6da5e-108">If an <xref:System.AggregateException?displayProperty=name> was being caught as a means of detecting a task that was cancelled prior to the WaitAll call being invoked, that code should instead do the same detection via the IsCanceled property (for example: .Any(t =&gt; t.IsCanceled)) since .NET 4.6 will only throw in that case if all awaited tasks are completed prior to the timeout.</span></span>|
|<span data-ttu-id="6da5e-109">スコープ</span><span class="sxs-lookup"><span data-stu-id="6da5e-109">Scope</span></span>|<span data-ttu-id="6da5e-110">マイナー</span><span class="sxs-lookup"><span data-stu-id="6da5e-110">Minor</span></span>|
|<span data-ttu-id="6da5e-111">Version</span><span class="sxs-lookup"><span data-stu-id="6da5e-111">Version</span></span>|<span data-ttu-id="6da5e-112">4.5</span><span class="sxs-lookup"><span data-stu-id="6da5e-112">4.5</span></span>|
|<span data-ttu-id="6da5e-113">型</span><span class="sxs-lookup"><span data-stu-id="6da5e-113">Type</span></span>|<span data-ttu-id="6da5e-114">ランタイム</span><span class="sxs-lookup"><span data-stu-id="6da5e-114">Runtime</span></span>|
|<span data-ttu-id="6da5e-115">影響を受ける API</span><span class="sxs-lookup"><span data-stu-id="6da5e-115">Affected APIs</span></span>|<ul><li><xref:System.Threading.Tasks.Task.WaitAll(System.Threading.Tasks.Task[],System.Int32)?displayProperty=nameWithType></li><li><xref:System.Threading.Tasks.Task.WaitAll(System.Threading.Tasks.Task[],System.Int32,System.Threading.CancellationToken)?displayProperty=nameWithType></li><li><xref:System.Threading.Tasks.Task.WaitAll(System.Threading.Tasks.Task[],System.TimeSpan)?displayProperty=nameWithType></li></ul>|

